wix/Build.OpenJDK_generic.cmd

```diff
@@ -36,55 +36,28 @@ IF NOT DEFINED PRODUCT_HELP_LINK SET PRODUCT_HELP_LINK=https://github.com/adopti
 IF NOT DEFINED PRODUCT_SUPPORT_LINK SET PRODUCT_SUPPORT_LINK=https://adoptium.net/support
 IF NOT DEFINED PRODUCT_UPDATE_INFO_LINK SET PRODUCT_UPDATE_INFO_LINK=https://adoptium.net/temurin/releases
 
-REM This needs tidying up, it's got out of control now
-IF NOT "%ARCH%" == "x64" (
-	IF NOT "%ARCH%" == "x86-32" (
-        IF NOT "%ARCH%" == "arm64" (
-            IF NOT "%ARCH%" == "x86-32 x64" (
-                IF NOT "%ARCH%" == "x86-32 arm64" (
-                    IF NOT "%ARCH%" == "arm64 x86-32" (
-                        IF NOT "%ARCH%" == "x64 x86-32" (
-                            IF NOT "%ARCH%" == "x64 arm64" (
-                                IF NOT "%ARCH%" == "arm64 x64" (
-                                    IF NOT "%ARCH%" == "x86-32 x64 arm64" (
-                                        IF NOT "%ARCH%" == "x86-32 arm64 x64" (
-                                            IF NOT "%ARCH%" == "arm64 x86-32 x64" (
-                                                IF NOT "%ARCH%" == "arm64 x64 x86-32" (
-                                                    IF NOT "%ARCH%" == "x86-32 x64 arm64" (
-                                                        IF NOT "%ARCH%" == "x64 x86-32 arm64" (
-                                                            ECHO ARCH %ARCH% not supported : valid values : x64, x86-32, arm64, x86-32 x64, x64 x86-32, x86-32 x64 arm64, x86-32 arm64 x64, arm64 x86-32 x64, arm64 x64 x86-32, x86-32 x64 arm64, x64 x86-32 arm64
-                                                            GOTO FAILED
-                                                        )
-                                                    )
-                                                )
-                                            )
-                                        )
-                                    )
-                                )
-                            )
-                        )
-                    )
-                )
-            )
-        )
-	)
+powershell -ExecutionPolicy Bypass -File "%~dp0\helpers\Validate-Input.ps1" ^
+    -toValidate '%ARCH%' ^
+    -validInputs "x64 x86-32 x86 arm64" ^
+    -delimiter " "
+
+IF %ERRORLEVEL% == 1 (
+    ECHO ARCH %ARCH% not supported : valid values are any combination of : x64, x86-32, arm64
```

Comment:
"x86" is missing here.

(I don't know how x86 is different from x86-32 and why it as been added)

Reply:
`x86` is the same as `x86-32`. Some vendors (like Microsoft) use `x86` instead of `x86-32` so this change helps vendors use this code without local modification

Reply:
Note, I updated this for you in my latest PR:
https://github.com/adoptium/installer/pull/851